client/hosting/sites/components/sites-dashboard.tsx

```diff
@@ -98,10 +100,20 @@ const SitesDashboard = ( {
 	const [ initialSortApplied, setInitialSortApplied ] = useState( false );
 
 	const { hasSitesSortingPreferenceLoaded, sitesSorting, onSitesSortingChange } = useSitesSorting();
+	const sitesFilterCallback = ( site: SiteExcerptData ) => {
+		return ! site.options?.is_domain_only;
+	};
+
+	const getFilterArgs = ( siteType: string ): string[] => {
+		if ( siteType === 'all' ) {
+			return [ 'non-p2' ];
+		}
+		return [ 'p2' ];
+	};
 
 	const { data: allSites = [], isLoading } = useSiteExcerptsQuery(
-		[],
-		( site ) => ! site.options?.is_domain_only,
+		getFilterArgs( siteType ),
```

Comment:
This works, but I wonder is it better to query all sites regardless of whether we are on /p2s or /sites, and just filter what is returned from the endpoint differently depending on where we are? That way all the data is there when switching between the two initially, and the user doesn't have to wait for a query to happen on each dashboard? 🤔 

Maybe we could get away with something like that if we keep the fetch the same in all cases (so go back to [] for the fetchFilter), and just include the p2/non-p2 check in the sitesFilterCallback? I assume that would allow the underlying useQuery to just pull data it already has since all the params for the request itself are the same in that case, as opposed to making a new query.

I suppose there are cons to this approach as well, but it seems like it might have a few benefits for us.

Reply:
I'm not sure I need the `p2`/`non-p2` filter, I think the better solution would be to just make the one get sites query and show the p2/non-p2 sites depending on what dashboard user is looking at. I will look into this.

Reply:
No longer need to make separate get sites queries for p2/non-p2 - this filtering is done in `useSiteExcerptsQuery`